Updates to KES docs for config reference (#48)
minio/kes-docs: content/_includes/server-config.md

# The API configuration. The APIs exposed by the KES server can
# be adjusted here. Each API is identified by its API path.
#

> # In general, the KES server uses sane defaults for all APIs.

Reviewer: Maybe a copy/paste error from somewhere, I recall we fixed this a while back
```suggestion
# In general, the KES server uses reasonable defaults for all APIs.
```

Author: I'll go ahead and change it, but i copy/paste this from the KES repo. It'll need to change over there for long-term effects. Because I will forget to change it here every time.

# Only customize the APIs if there is a real need.
# 
# Disabling authentication for an API must be carefully evaluated.

> # One example, when disabling authentication may be justified, would
> # be the liveness and readiness probes in a Kubernetes environment.

Reviewer: maybe
```suggestion
# Possible examples are the liveness and
# readiness probes in a Kubernetes environment.
```

Author: Suggest we make this change in the KES repo.

      engine: ""  # The path of the transit engine - e.g. "my-transit". If empty, defaults to: transit (Vault default)
      key: ""     # The key name that should be used to encrypt entries stored on the K/V engine.
    approle:    # AppRole credentials. See: https://www.vaultproject.io/docs/auth/approle.html

>       namespace: "" # Optional Vault namespace used just for authentication. A single "/" is an alias for the Vault root namespace.

Reviewer: Idea:
```suggestion
      namespace: "" # Optional Vault namespace used only for authentication. For the Vault root namespace, use "/".
```
I'm not sure if it is significant that `/` is an alias, ignore if I have it wrong.

Author: 👍

>       engine: ""    # The path of the AppRole engine - e.g. authenticate. If empty, defaults to: approle. (Vault default)

Reviewer: I don't know what `path` is here, but `path of` something that doesn't look like a file path feels odd. Also maybe replace `e.g.` with something along the lines of `for example, "authenticate".`

Author: 👍

      id: ""        # Your AppRole Role ID
      secret: ""    # Your AppRole Secret ID
    kubernetes: # Kubernetes credentials. See: https://www.vaultproject.io/docs/auth/kubernetes
      namespace: "" # Optional Vault namespace used just for authentication. A single "/" is an alias for the Vault root namespace.

>       engine: ""    # The path of the Kubernetes engine e.g. authenticate. If empty, defaults to: kubernetes. (Vault default)

Reviewer: Same comment as above.

Author: 👍

      # with Azure managed credentials.
      managed_identity:
        client_id: ""      # The Azure managed identity of the client - i.e. a UUID.

  entrust:
    # The Entrust KeyControl configuration.
    # For more information take a look at:
    # https://www.entrust.com/digital-security/key-management/keycontrol
    keycontrol:
      endpoint: ""     # The KeyControl endpoint - e.g. https://keycontrol.my-org.com
      vault_id: ""     # The Vault ID            - e.g. e30497c1-bff7-4e81-beb7-fb35c4b7410c
      box_id:   ""     # The Box name or ID      - e.g. tenant-1
      # The KeyControl access credentials
      credentials:

>         username: ""   # The username able to access the Vault and Box.

Reviewer: maybe
```suggestion
        username: ""   # A username with access to the Vault and Box.
```

Also more `e.g.` in this area that could be replaced with words

Author: Replaced those and `i.e.` throughout. But will need to also address in the KES repo.

        password: ""   # The user password
      # The KeyControl client TLS configuration
      tls:

>         ca: ""         # Path to one or multiple PEM-encoded CA certificates for verifying the KeyControl TLS certificate.

Reviewer: ```suggestion
        ca: ""         # Path to one or more PEM-encoded CA certificates for verifying the KeyControl TLS certificate.
```
